File: interactiveTransition的副本/interactiveTransition/ViewController.m
//
//  ViewController.m
//  interactiveTransition
//
//  Created by kfz on 2017/2/24.
//  Copyright © 2017年 kongfz. All rights reserved.
//

#import "ViewController.h"
#import "BViewController.h"
#import "KFZTransitionAnimation.h"
#import "SwipeUpInteractiveTransition.h"


@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
}

- (IBAction)presentVC:(id)sender {
    BViewController *bVC = [[BViewController alloc] init];
    
    
    [self presentViewController:bVC animated:YES completion:nil];
}

//- (nullable id <UIViewControllerAnimatedTransitioning>)animationControllerForPresentedController:(UIViewController *)presented presentingController:(UIViewController *)presenting sourceController:(UIViewController *)source {
//    return self.transitionAnimation;
//}
//
//- (nullable id <UIViewControllerAnimatedTransitioning>)animationControllerForDismissedController:(UIViewController *)dismissed {
//    return self.transitionAnimation;
////    return [NormalDismissAnimation new];
//}
//
//- (nullable id <UIViewControllerInteractiveTransitioning>)interactionControllerForDismissal:(id <UIViewControllerAnimatedTransitioning>)animator {
//    return self.interactiveTransition.interacting ? self.interactiveTransition : nil;
//}




@end

File: navigationTitle_bug/navigationTitle_bug/ViewController.swift
//
//  ViewController.swift
//  navigationTitle_bug
//
//  Created by kfz on 16/11/12.
//  Copyright © 2016年 kongfz. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        title = NSStringFromClass(self.classForCoder)
        
        title = NSStringFromClass(self.classForCoder)
        
        let appName: String = Bundle.main.object(forInfoDictionaryKey: "CFBundleName") as! String
        print(appName)
        
//        let c: = NSClassFromString(appName+".BViewController")
        
    }

    
    @IBAction func pushToAVC(_ sender: Any) {
        let avc = AViewController()
        navigationController?.pushViewController(avc, animated: true)
        
    }
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        navigationController?.isNavigationBarHidden = true
    }
    
    override func viewWillDisappear(_ animated: Bool) {
        super.viewWillDisappear(animated)
        navigationController?.isNavigationBarHidden = false
    }


}

